test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+    // a mocha bug where `afterEach` hooks aren't run for skipped tests.
+    // https://github.com/mochajs/mocha/issues/2546
+    sandbox.restore();
+    // Prevent logs in the mocha output.
+    sandbox.stub(logger, 'warn');
+    sandbox.stub(logger, 'log');
+  });
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe(`constructor`, function() {
+    it(`should construct without any inputs`, function() {
+      expect(() => {
+        new Router();
+      }).to.not.throw();
+    });
+  });
+
+  describe(`calling addFetchListener()`, function() {
+    it(`should return true and add a fetch listener the first time addFetchListener() is called`, function() {
+      const spy = sandbox.spy(self, 'addEventListener');
+
+      const router = new Router();
+      const added = router.addFetchListener();
+
+      expect(added).to.be.true;
+      expect(spy.calledOnce).to.be.true;
+      expect(spy.firstCall.args[0]).to.eql('fetch');
+      expect(spy.firstCall.args[1]).to.be.a('function');
+    });
+
+    it(`should return false and not add a fetch listener the second time addFetchListener() is called`, function() {
+      const spy = sandbox.spy(self, 'addEventListener');
+
+      const router = new Router();
+      const addedFirst = router.addFetchListener();
+      const addedSecond = router.addFetchListener();
+
+      expect(addedFirst).to.be.true;
+      expect(addedSecond).to.be.false;
+      expect(spy.calledOnce).to.be.true;
+    });
+  });
+
+  describe(`registering and unregistering routes`, function() {
+    it(`should throw in dev when registerRoute is called with a parameter that doesn't match the expected interface`, async function() {
+      if (process.env.NODE_ENV === 'production') return this.skip();
+
+      const router = new Router();
+
+      await expectError(
+        () => router.registerRoute({}),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoute({handle: HANDLER, method: METHOD}),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoute({match: MATCH, method: METHOD}),
+        'incorrect-type',
```

Comment:
Not sure why this is an incorrect type. I was expecting a missing method because there is no handle method.